rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6552,299 @@ private void assertRefundInProcessPegInVersionLegacy(
+    public void registerFastBridgeBtcTransaction__after_RSKIP293_activation() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+        Coin valueToSend = Coin.COIN;
+
+        BigInteger result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            true,
+            true,
+            true
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend.multiply(2)).asBigInteger(), result);
+    }
+
+    @Test
+    public void registerFastBridgeBtcTransaction_no_retiring_federation() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+        Coin valueToSend = Coin.COIN;
+
+        BigInteger result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            true,
+            false,
+            false
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend).asBigInteger(), result);
+
+        // test when amount is sent to the active federation and also to a retiring federation that doesn't exist
+        result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            true,
+            true,
+            false
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend).asBigInteger(), result);
+
+        // test when amount is sent to a retiring federation that doesn't exist
+        result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            false,
+            true,
+            false
+        );
+        Assert.assertEquals(FAST_BRIDGE_UNPROCESSABLE_TX_VALUE_ZERO_ERROR, result.longValue());
+    }
+
+    @Test
+    public void registerFastBridgeBtcTransaction_funds_sent_to_retiring_federation() throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP176)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+        Coin valueToSend = Coin.COIN;
+
+        BigInteger result = registerFastBridgeBtcTransaction_RSKIP293(
+            activations,
+            valueToSend,
+            true,
+            false,
+            false
+        );
+        Assert.assertEquals(co.rsk.core.Coin.fromBitcoin(valueToSend).asBigInteger(), result);
```

Comment:
This is not sending to the retiring federation

Reply:
Done